process/factory/interceptorscontainer/shardInterceptorsContainerFactory.go

```diff
@@ -76,6 +76,12 @@ func NewShardInterceptorsContainerFactory(
 	if check.IfNil(args.EpochStartTrigger) {
 		return nil, process.ErrNilEpochStartTrigger
 	}
+	if len(args.ChainID) == 0 {
+		return nil, process.ErrInvalidChainID
```

Comment:
Please treat those 2 errors in multidata and single data interceptor and blacklist the fromconnected peer and message peer originator

Reply:
In multidatainterceptor.go, L186 and in singledatainterceptor.go L95 please add:
```GO
		isWrongVersion := err == process.ErrInvalidTransactionVersion || err == process.ErrInvalidChainID
		if isWrongVersion{
			//this situation is so severe that we need to black list de peers
			reason := "wrong version of received intercepted data, topic " + mdi.topic + ", error " + err.Error()
			mdi.antifloodHandler.BlacklistPeer(originator, reason, core.InvalidMessageBlacklistDuration)
			mdi.antifloodHandler.BlacklistPeer(fromConnectedPeer, reason, core.InvalidMessageBlacklistDuration)
		} 
```

please add tests for these condition. Thanks

Reply:
done.